adagrams/game.py

```diff
@@ -1,11 +1,142 @@
+import random
+
+LETTER_POOL = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
+
+SCORE_CHART = {
+    'A': 1,
+    'B': 3,
+    'C': 3,
+    'D': 2,
+    'E': 1,
+    'F': 4,
+    'G': 2,
+    'H': 4,
+    'I': 1,
+    'J': 8,
+    'K': 5,
+    'L': 1,
+    'M': 3,
+    'N': 1,
+    'O': 1,
+    'P': 3,
+    'Q': 10,
+    'R': 1,
+    'S': 1,
+    'T': 1,
+    'U': 1,
+    'V': 4,
+    'W': 4,
+    'X': 8,
+    'Y': 4,
+    'Z': 10
+}
+
+
 def draw_letters():
-    pass
+    ## pool_of_letters_array is an array that will store the letters and the number of times it occurs
+    pool_of_letters_array = []
+
+    ## for loop: iteration through LETTER_POOL to combine subset list into the pool_of_letters_array list
+    for letter in LETTER_POOL: 
+        pool_of_letters_array += list(letter * LETTER_POOL[letter])
```

Comment:
Hi @kelsey-steven-ada, this is a very interesting read and, after reading, I do recall something from one of the lessons around lists and memory. Correct me if I'm wrong but essentially, in the blog's example, the use of the multiplication operator in a 2D list would create a shallow copy however many times it's multiplied -- making changes to an element in one of those lists would be reflected in the other lists too because the lists all reference the same memory

Reply:
Exactly! If our list is filled with immutable data types like ints, it isn't really an issue. But the moment we have mutable objects like a list of lists or a list of dictionaries, it can cause big problems if the shared memory isn't intended.